### DISCUSSION TITLE
Migrating from 1.8.5 to 2.3.2

### DISCUSSION THREAD

**Discussion by amd-9** [recharts/recharts#3174] [Q&A] [answered]
Is there any guide on migrating from 1.8.5 to 2.3.2?

**Comment by ckifer** [comment]
I don't think there is currently but this would be a good thing to have. We can maybe add a minimal guide here and move it from here to the website

**Comment by ckifer** [comment, marked as answer]
From 1.8.5 to 2.0.0:

* Recharts has been refactored to TypeScript - you may run into random typing issues. Fix properties that the compiler complains about and update to the correct type.
* Drop support for React < 16


Looking at the changelog its _really really_ hard to spot breaking changes as they aren't called out at all. The upgrade to 2.0 was mostly to use TypeScript. Type issues will probably be the biggest thing you run into.

From 2.0.0 to 2.3.2:
All changes here in minor and patch versions are _supposed_ to be non-breaking, but some things have been merged that are breaking over time.
 
Items to watch out for:
  * Tooltip tab index and pie chart tab index issue (adds border on tooltip, makes the tooltip auto-focus)
  * Tooltip formatter types broken with `TValue` 

Best bet is to upgrade yourself and then document what you need to do here. Sorry my answer this isn't better than this 😅 . Please please document any upgrades here! Thanks

**Comment by seiyab** [comment]
I'm migrating now and have a question.
Sometimes I want to use Recharts's (internal?) types like `Coordinate`.
IDE suggests importing them from modules such as `'recharts/types/util/types'`.
Are they exported intentionally, and won't change frequently?
Or, should we avoid importing them?

**Reply by ckifer** [reply]
Our tenants will be to not introduce breaking changes (including with types) without a new major version. I'd stay as top-level as you can `recharts/types` or the types exported at `recharts` - but we'll try not to move things around randomly.

**Reply by seiyab** [reply]
Thank  you for your response. I'd avoid relying types that is not exposed at top-level.

**Comment by seiyab** [comment]
Just for information to who are planning to migrate to v2.
V2 has https://github.com/recharts/recharts/issues/2176.
We are stopping migration because of this.

**Reply by ckifer** [reply]
Can you create an updated code sandbox that recreates that issue?

**Reply by seiyab** [reply]
Here it is.
https://codesandbox.io/s/serene-lichterman-g0s9t6?file=/src/Repro.tsx

**Reply by seiyab** [reply]
I also pasted screenshot at the issue comment https://github.com/recharts/recharts/issues/2176#issuecomment-1564085665.
